### PR TITLE
Improve Search Times

### PR DESCRIPTION
**Pull request by silent-po2**
During load testing it was found that our database searches take considerably long when using searches that "hit" hundreds of data sets, with this being around a whole minute in the worst case.

As such times are not acceptable, #292 intends to rectify this by reworking the backend of search to reduce the amount of queries ran against the database during a search.

### CODE REVIEW

File: Server/src/models/DatasetQueryModel.ts [hunk 1]
```diff
@@ -276,4 +261,9 @@ export class DataQueryModel {
             dataPoint.values = typeof dataPoint.values === 'string' ? JSON.parse(dataPoint.values) : []
         });
     }
+    private valuesToArray2 = (datapointComments: any): void => {
```
Comment (ds161833) [inline Server/src/models/DatasetQueryModel.ts]: poor name, i suggest parseDatapointComments()

File: Server/src/services/getDatasetService.ts [hunk 2]
```diff
@@ -167,25 +168,107 @@ export class retrieveData {
...
+                    dataPointComments: dataPointComments
+                }
+
+                allDataSets.push(singleDataSet);
             }
-            return setOfData
+            return allDataSets
         } catch (error) {
-            console.error(error);
+            throw new InternalServerError("Something went wrong fetching from DB. Maybe its down")
```
Comment (ds161833) [inline Server/src/services/getDatasetService.ts:274]: nice

**Review by ds161833 — APPROVED** [review]

File: Server/src/models/DatasetQueryModel.ts (hunk 1 above)
Comment (silent-po2) [reply Server/src/models/DatasetQueryModel.ts]: Good catch will fix

File: Server/src/routes/getDatasetRouter.ts [hunk 3]
```diff
@@ -27,7 +27,7 @@ router.delete('/api/v1/dataset/removeSavedDatset/:userEmail/:datasetId', (reques
     getDataControllerObject.createRequestForRemovingSavedDataset(request, response);
 });
 
-router.get('/dataset*', (request: Request, response: Response) => {
+router.get('/api/v1/dataset*', (request: Request, response: Response) => {
```
Comment (GD-Dheer) [inline Server/src/routes/getDatasetRouter.ts:30]: Nice. Reminder to change in FE if this is updated here

**Review by GD-Dheer — APPROVED** [review]
LGTM. Thanks for adding these performance improvements

File: Server/src/routes/getDatasetRouter.ts (hunk 3 above)
Comment (silent-po2) [reply Server/src/routes/getDatasetRouter.ts:30]: Good catch; done

**Pull request merged**